src/zope/viewlet/viewlet.py

```diff
@@ -96,6 +104,11 @@ class ResourceViewletBase(object):
     _path = None
 
     def getURL(self):
+        """
+        Lookup the resource from our path using the
+        :class:`++resource++ namespace <zope.traversing.namespace.resource>`
+        and call it, returning the results.
```

Comment:
"The results" will be the URL of that resource, since that's what you get when you call resources.

Reply:
To nitpick a bit, that may be what you get when you call a `zope.browserresource.interfaces.IResource`, but that's not guaranteed by the (default) ++resource namespace. The (default) resource namespace, as implemented in `zope.traversing.namespace.resource`, simply finds the default named adapter for the request. The interface that adapter implements is not specified. *This* object puts a further (previously implicit) constraint upon it that it is callable. This object doesn't even require that the results of calling it be a string.

This object is further derived with objects like JavaScriptViewlet that use templates and assume that getURL returns something that is usable as a string in a TALES context. Perhaps those objects could use more documentation, and **certainly** the common usage and interaction with zope.browserresource could also be called out more clearly.

Multiple layers of that process can be overridden by other frameworks, starting with different/more specific registrations for ++resource. One framework that I work on doesn't use zope.browserresource at all, but does have objects that we use as resources.

(This is neither here nor there, but the reason I've been working on a lot of this documentation lately is that I've been implementing a [ZPT-based template system](http://ntinikola-chameleon.readthedocs.io/en/latest/) for the [nikola static website generator](https://getnikola.com/handbook.html). It's outside of any Zope3 framework, so I had to put a lot of these pieces together again for myself as I wrote and documented it; I'm *still* learning things as I help with this documentation---and your comments have been very helpful.)